### DISCUSSION TITLE
Report parts that need to be ordered

### DISCUSSION THREAD

**Discussion by JustusRijke** [inventree/InvenTree#2383] [Q&A] [answered]
First of all thank you for your efforts in developing InvenTree!
We are trying out InvenTree for our small production facility to get a better overview on parts, in order to reduce the risk of ordering too late (understock) or too soon (overstock). 

But I couldn't find a way to get a report of the parts we need to order (other than exporting all parts to Excel and filtering/calculating etc.).

I've created parts & created a build order. On the "parts" overview page, all parts are listed with the current stock amount but no column with the quantity that needs to be ordered.

So I thought about creating a custom report functionality (there are many part context variables that can be used). But the admin page doesn't show a Parts category:
![image](https://user-images.githubusercontent.com/53965859/143781943-69d06bcd-5cfb-4469-8d7f-f14c2c8907d7.png)

I am probably overlooking something, suggestions are welcome 🙂

**Comment by SchrodingersGat** [comment, marked as answer]
On the index page there is a "required for build orders" tab here:

![image](https://user-images.githubusercontent.com/10080325/143943552-6ac22c9b-50f8-4435-ab74-7e6ab1b5f796.png)

Does this provide the functionality you are looking for?

Or are you looking to export a specific document? PDF or excel?

**Reply by JustusRijke** [reply]
That page indeed list the parts that are needed, but not the amount we need to order to reach minimum stock level + stock for the build orders. Ideally this would be available via a report (Excel, PDF, printable web page) that also shows the current stock level so we can validate stock level before ordering.

![image](https://user-images.githubusercontent.com/53965859/144202947-5d8a4e9a-ebe3-427c-bcfa-9c7d25a59595.png)

**Reply by matmair** [reply]
@JustusRijke this would be a new feature - I created an FR (#2399) for it. Feel free to add more detail if you want.
As I can see from the Category names you have some language knowledge - if you have time we always appreciate contributions to our [translations](https://crowdin.com/project/inventree) so people with lesser knowledge of English can also use the project.

**Reply by JustusRijke** [reply]
Great! thanks for creating the FR. Will look into helping out with the translations, good suggestion.

To reach my goal for now, I suppose the best option now is to use the Excel export function and go from there?

**Reply by matmair** [reply]
Probably, will look into how much effort it is to implement this tonight

**Reply by JustusRijke** [reply]
Much appreciated! I started on the translations. My first time working with crowdin.com, so just be sure can you let me know the translations came through?

**Reply by matmair** [reply]
@JustusRijke your 55 translations came through. They are pending for approval of a proofreader. If nobody does that in the next week or two I will do it. Thanks for your submissions, good translations are quite hard to get by for open source projects!

**Reply by JustusRijke** [reply]
1. Created an assembly with some purchasable parts in its BOM
2. Added the assembly to a build order

On the "Required for build orders" it shows the assembly. But (as expected) not the BOM parts of the assembly.
Now, Is there a way to find the (quantity of) purchasable parts required for this build order? 
This data is not available in the parts Excel export AFAIK.

**Reply by matmair** [reply]
@JustusRijke there is a button for that in the 'Pending Items' tab: The cart button allows you to order the needed item
![grafik](https://user-images.githubusercontent.com/66015116/144751962-65f05176-d05d-4b70-bbbd-b8b6c4385f73.png)

**Reply by JustusRijke** [reply]
@matmair that looks promising, where can I find the 'Pending Items' tab? note that I am running on the stable branch (0.5.4)

**Reply by matmair** [reply]
@JustusRijke I do not have access to a 0.5.4 instance rn, in master it currently is in the order tab. You have to add a build output before this table appears. As far as I can remember this should be there in 0.5.x as well.

**Reply by matmair** [reply]
I will mark the answer above as valid again, if anything else comes up, please open a new discussion. The docs section for orders is here: https://inventree.readthedocs.io/en/latest/build/build/

**Comment by JustusRijke** [comment]
@matmair added a build output, but no table appears. The help pages do not mention a Pending Items tab. Where should this table appear, e.g. InvenTree Admin -> BUILD -> Build Orders? or somewhere in the normal user interface, like:
![image](https://user-images.githubusercontent.com/53965859/144796452-6b8d8f22-991e-4899-96be-eb1d4fa5da0e.png)

**Reply by matmair** [reply]
> @matmair added a build output, but no table appears. The help pages do not mention a Pending Items tab.

I do not have ready access to a 0.5.4 interface but it should be the allocate tab then. It couldbe that the better allocation workflow was just introduced in 0.6.x @SchrodingersGat ?

> Where should this table appear, e.g. InvenTree Admin -> BUILD -> Build Orders? or somewhere in the normal user interface, like:
> 
> 
 Please do not use the admin interface for anything regarding InvenTree regularly, it gives you unchecked access and lets you wreck your objects.

**Reply by JustusRijke** [reply]
Ok found it. Compared 0.5.4 with 0.6.x (master):
0.5.4 has a tab Build Output, showing completed / incomplete build outputs.
0.6.x has a tab Pending Items, showing incomplete build outputs, and a tab Completed Items, showing completed build outputs.
